src/llama.cpp

```diff
@@ -3997,9 +3997,6 @@ struct llm_build_context {
     struct ggml_cgraph * build_llama() {
         struct ggml_cgraph * gf = ggml_new_graph_custom(ctx0, llama_model_max_nodes(model), false);
 
-        // mutable variable, needed during the last layer of the computation to skip unused tokens
-        int32_t n_tokens = this->n_tokens;
-
```

Comment:
This variable is mutated at `il == n_layer - 1`, which IIUC has practically no effect on the computation. Am I missing something? Maybe the intend was to mutate it at `il == n_layer - 2`?

Reply:
Seems like historically it was used by `llm_build_moe_ffn`, but was replaced by `int64_t n_tokens = cur->ne[1]` at some point. It's safe to remove it here though